src/features/Dashboard/InstructorAssignSection/index.jsx

```diff
@@ -47,13 +50,22 @@ const InstructorAssignSection = () => {
     <Row>
       <Col xs="12">
         <h4 className="title-instr-assign">Instructor assignment</h4>
-        {classCards.map(classInfo => <ClassCard data={classInfo} key={classInfo?.classId} />)}
-        {classesData.length > numberOfClasses && (
+        {isLoadingClasses && (
+          <div className="w-100 h-100 d-flex justify-content-center align-items-center">
+            <Spinner
+              animation="border"
+              className="mie-3"
+              screenReaderText="loading"
+            />
+          </div>
+        )}
+        {!isLoadingClasses && classCards.map(classInfo => <ClassCard data={classInfo} key={classInfo?.classId} />)}
```

Comment:
Great addition, but I would like to suggest that changes outside the scope of the issue should go into a separate PR. This way, changes won't get mixed, and it will make the review process easier by avoiding unrelated updates to the issue's scope. Additionally, if a revert is needed, it would only affect a specific change rather than multiple changes at once.